diagnostics/buffer.go

```diff
@@ -0,0 +1,124 @@
+package diagnostics
+
+import (
+	"sync/atomic"
+
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/topdown"
+)
+
+var seq uint64
+
+// Buffer stores diagnostic information. They are not safe for concurrent access.
+type Buffer struct {
```

Comment:
Did you consider using Go's `container/ring` package? Anyway, if we don't use that package, I think this should be made private/internal to the server. I don't see a reason to expose it to users.

As a side note, if this file is kept, it needs:

- Copyright boilerplate
- Package comment